### ISSUE TITLE
[Feat] Data comparator shouldn't be a re-render criteria 

### ISSUE THREAD

**Issue by ricardoekm** [visgl/deck.gl#6660]
### Description

We've implemented the dataComparator feature in order to optimize our deck.gl application, however we noticed an unexpected behavior. Even if the data didn't change the layer was being re-rendered. 

Investigating, this happens because the dataComparator itself is considered in the re-render process. An implementation similar to the below will always trigger a re-render because the data comparator is always different (a new arrow function):

```javascript
   dataComparator: () => true
```

Similar to the accessors the dataComparator shouldn't be considered in the re-render process.

Thanks.

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Steps to Reproduce

```javascript
   dataComparator: () => true
```

### Environment

- Framework version: 8.6.8
- Browser: Chrome
- OS: MacOS

**Comment by Pessimistress** [comment]
Thanks for reporting. Fixed in v8.6.9.

**Comment by ricardoekm** [comment]
Great, thanks!
